### PR TITLE
Cors Pre-fligth

### PR DESCRIPTION
**Pull request by maneta**
Introduces CORS pre flight support

### CODE REVIEW

File: src/cors-proxy/init.lua [hunk 1]
```diff
@@ -47,31 +47,58 @@ local api_docs_headers = {
...
+  ngx.header['Access-Control-Allow-Credentials'] = 'true'
+end
+
+local function cors_preflight_response()
+  set_cors_headers()
+  ngx.status = 204
+  ngx.exit(ngx.status)
+end
+
+local function cors_preflight_response()
```
Comment (mikz) [inline src/cors-proxy/init.lua]: This method is duplicated.

Comment (maneta) [reply src/cors-proxy/init.lua]: oops :S  

Solved

File: src/cors-proxy/init.lua [hunk 2]
```diff
@@ -47,31 +47,52 @@ local api_docs_headers = {
...
-  ngx.var.proxy_path = upstream.path
-  ngx.req.set_uri_args(upstream.args)
+local function cors_preflight_response()
+  set_cors_headers()
+  ngx.status = 204
+  ngx.exit(ngx.status)
+end
+
+function _M:rewrite()
+  if cors_preflight() then
```
Comment (mikz) [inline src/cors-proxy/init.lua]: Have you tried this? This can't work.

Comment (maneta) [reply src/cors-proxy/init.lua]: Yes sorry about that, I was checking it and I mixed up copying the stuff.. gimme 5 minutes

**Comment by maneta** [comment]
Finally @mikz `tested`

**Review by mikz — APPROVED** [review]
@maneta trusted 👍

**Pull request merged**